### DISCUSSION TITLE
Code generators not working anymore. Lots of compiler warnings. Version 2.4.0.2960

### DISCUSSION THREAD

**Discussion by dxt-rr** [shinyorg/shiny#823] [Q&A] [answered]
Hello there,

I wanted to update Shiny to the latest version but it seems it doesn't work anymore...

1. I get tons of warnings regarding the code generators (MS SDK, VS 2019, XF are all up to date)
`CS8032: Eine Instanz des Shiny.Generators.StaticClassSourceGenerator-Analyzers kann nicht aus C:\Users\MYSELF\.nuget\packages\shiny\2.4.0.2960\analyzers\dotnet\cs\Shiny.Generators.dll erstellt werden: Die Datei oder Assembly "Microsoft.CodeAnalysis, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35" oder eine Abhängigkeit davon wurde nicht gefunden. Das System kann die angegebene Datei nicht finden..
CS8032: Eine Instanz des Shiny.Generators.AndroidApplicationSourceGenerator-Analyzers kann nicht aus C:\Users\MYSELF\.nuget\packages\shiny\2.4.0.2960\analyzers\dotnet\cs\Shiny.Generators.dll erstellt werden: Die Datei oder Assembly "Microsoft.CodeAnalysis, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35" oder eine Abhängigkeit davon wurde nicht gefunden. Das System kann die angegebene Datei nicht finden..
CS8032: Eine Instanz des Shiny.Generators.AndroidActivitySourceGenerator-Analyzers kann nicht aus C:\Users\MYSELF\.nuget\packages\shiny\2.4.0.2960\analyzers\dotnet\cs\Shiny.Generators.dll erstellt werden: Die Datei oder Assembly "Microsoft.CodeAnalysis, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35" oder eine Abhängigkeit davon wurde nicht gefunden. Das System kann die angegebene Datei nicht finden..
CS8032: Eine Instanz des Shiny.Generators.iOSAppDelegateSourceGenerator-Analyzers kann nicht aus C:\Users\MYSELF\.nuget\packages\shiny\2.4.0.2960\analyzers\dotnet\cs\Shiny.Generators.dll erstellt werden: Die Datei oder Assembly "Microsoft.CodeAnalysis, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35" oder eine Abhängigkeit davon wurde nicht gefunden. Das System kann die angegebene Datei nicht finden..
CS8032: Eine Instanz des Shiny.Generators.UwpSourceGenerator-Analyzers kann nicht aus C:\Users\MYSELF\.nuget\packages\shiny\2.4.0.2960\analyzers\dotnet\cs\Shiny.Generators.dll erstellt werden: Die Datei oder Assembly "Microsoft.CodeAnalysis, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35" oder eine Abhängigkeit davon wurde nicht gefunden. Das System kann die angegebene Datei nicht finden..`

I have noticed that several samples have been changed to the "Boilerplate-NoCodeGen" schema. So I guess it may be a general bug. I have no problem doing the boilerplate stuff, but I would like to get rid of the warnings. How can I disable Shiny code generation at all?

2. As soon I install the Shiny nuget package I get multiple warnings regarding a version conflict in "System.Numerics.Vectors"
`Konflikt zwischen System.Numerics.Vectors, Version=2.0.5.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a und System.Numerics.Vectors, Version=4.1.3.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a.
    System.Numerics.Vectors, Version=2.0.5.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a wurde aufgrund der primären Eigenschaft System.Numerics.Vectors, Version=4.1.3.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a vorgezogen.
    Verweise, die von "System.Numerics.Vectors, Version=2.0.5.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" abhängig sind [C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\Common7\IDE\ReferenceAssemblies\Microsoft\Framework\Xamarin.iOS\v1.0\System.Numerics.Vectors.dll].
        C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\Common7\IDE\ReferenceAssemblies\Microsoft\Framework\Xamarin.iOS\v1.0\System.Numerics.Vectors.dll
          Elementincludes einer Projektdatei, die den Verweis "C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\Common7\IDE\ReferenceAssemblies\Microsoft\Framework\Xamarin.iOS\v1.0\System.Numerics.Vectors.dll" verursacht haben.
            System.Numerics.Vectors
    Verweise, die von "System.Numerics.Vectors, Version=4.1.3.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" abhängig sind [].
        C:\Users\MYSELF\.nuget\packages\system.memory\4.5.4\lib\netstandard2.0\System.Memory.dll
          Elementincludes einer Projektdatei, die den Verweis "C:\Users\MYSELF\.nuget\packages\system.memory\4.5.4\lib\netstandard2.0\System.Memory.dll" verursacht haben.
            C:\Users\MYSELF\.nuget\packages\system.memory\4.5.4\lib\netstandard2.0\System.Memory.dll
C:\Users\MYSELF\.nuget\packages\microsoft.extensions.logging.abstractions\6.0.0\lib\netstandard2.0\Microsoft.Extensions.Logging.Abstractions.dll
            C:\Users\MYSELF\.nuget\packages\microsoft.extensions.logging\6.0.0\lib\netstandard2.1\Microsoft.Extensions.Logging.dll
            C:\Users\MYSELF\.nuget\packages\microsoft.extensions.primitives\6.0.0\lib\netstandard2.0\Microsoft.Extensions.Primitives.dll
            C:\Users\MYSELF\.nuget\packages\microsoft.extensions.options\6.0.0\lib\netstandard2.1\Microsoft.Extensions.Options.dll
            C:\Users\MYSELF\.nuget\packages\shiny.core\2.4.0.2960\lib\xamarinios10\Shiny.Core.dll
C:\Users\MYSELF\.nuget\packages\system.diagnostics.diagnosticsource\6.0.0\lib\netstandard2.0\System.Diagnostics.DiagnosticSource.dll`

How can I solve that? I mean except staying with version 2.3.x of Shiny...

Thank's for help
Ralph

**Comment by aritchie** [comment, marked as answer]
I just tested them and it is working fine for me.  If you feel you've got an issue, you need to create a github issue with all of the appropriate details and a reproducible sample.  If you don't want codegen, remove the "Shiny" nuget and just reference Shiny.Core or the specific Shiny library you are using.

Also, please post any error messages in English if you want help with them.

**Reply by dxt-rr** [reply]
I was able to deactivate code generators as you suggested. Thank you for your advice.

I am still having the version conflicts though. I did set up a new machine with all Windows updates and latest VS 2019.
If I create a new Xamarin Forms app and update Forms and Essentials packages and then add Shiny.Core package the conflict appears. No matter if I add Shiny.Core to the default platform independend class library or add a new one like "Project.Shared".

I'm not sure, if this is an issue or if I missed to do anything. I just don't have an idea, what...
Sorry for the German error message. Here the one from the new machine, compiling the the location sample from 2.4:
`Found conflicts between different versions of "System.Numerics.Vectors" that could not be resolved.
There was a conflict between "System.Numerics.Vectors, Version=2.0.5.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" and "System.Numerics.Vectors, Version=4.1.3.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a".
    "System.Numerics.Vectors, Version=2.0.5.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" was chosen because it was primary and "System.Numerics.Vectors, Version=4.1.3.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" was not.
    References which depend on "System.Numerics.Vectors, Version=2.0.5.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" [C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\Common7\IDE\ReferenceAssemblies\Microsoft\Framework\Xamarin.iOS\v1.0\System.Numerics.Vectors.dll].
        C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\Common7\IDE\ReferenceAssemblies\Microsoft\Framework\Xamarin.iOS\v1.0\System.Numerics.Vectors.dll
          Project file item includes which caused reference "C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\Common7\IDE\ReferenceAssemblies\Microsoft\Framework\Xamarin.iOS\v1.0\System.Numerics.Vectors.dll".
            System.Numerics.Vectors
    References which depend on "System.Numerics.Vectors, Version=4.1.3.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" [].
        C:\Users\testuser\.nuget\packages\system.memory\4.5.4\lib\netstandard2.0\System.Memory.dll
          Project file item includes which caused reference "C:\Users\testuser\.nuget\packages\system.memory\4.5.4\lib\netstandard2.0\System.Memory.dll".
            C:\Users\testuser\.nuget\packages\system.memory\4.5.4\lib\netstandard2.0\System.Memory.dll
            C:\Users\testuser\.nuget\packages\microsoft.extensions.logging.abstractions\6.0.0\lib\netstandard2.0\Microsoft.Extensions.Logging.Abstractions.dll
            C:\Users\testuser\.nuget\packages\microsoft.extensions.logging\6.0.0\lib\netstandard2.1\Microsoft.Extensions.Logging.dll
            C:\Users\testuser\.nuget\packages\microsoft.extensions.primitives\6.0.0\lib\netstandard2.0\Microsoft.Extensions.Primitives.dll
            C:\Users\testuser\.nuget\packages\microsoft.extensions.options\6.0.0\lib\netstandard2.1\Microsoft.Extensions.Options.dll
            C:\Users\testuser\.nuget\packages\shiny.core\2.4.0.2960\lib\xamarinios10\Shiny.Core.dll
            C:\Users\testuser\.nuget\packages\shiny.locations\2.4.0.2960\lib\xamarinios10\Shiny.Locations.dll
            C:\Users\testuser\.nuget\packages\shiny.locations.common\2.4.0.2960\lib\xamarinios10\Shiny.Locations.Common.dll
            C:\Users\testuser\.nuget\packages\sqlite-net-pcl\1.8.116\lib\netstandard2.0\SQLite-net.dll
            C:\Users\testuser\.nuget\packages\sqlitepclraw.core\2.0.4\lib\netstandard2.0\SQLitePCLRaw.core.dll
            C:\Users\testuser\.nuget\packages\sqlitepclraw.bundle_green\2.0.4\lib\Xamarin.iOS10\SQLitePCLRaw.batteries_v2.dll
            C:\Users\testuser\.nuget\packages\sqlitepclraw.provider.dynamic_cdecl\2.0.4\lib\netstandard2.0\SQLitePCLRaw.provider.dynamic_cdecl.dll
            C:\Users\testuser\.nuget\packages\system.diagnostics.diagnosticsource\6.0.0\lib\netstandard2.0\System.Diagnostics.DiagnosticSource.dll`

**Comment by aritchie** [comment]
Ralph - please know that this Shiny is not a product, the samples come as is as shown on the readme (they aren't even fully done), and this is not technical support.  I give away all of the code, samples, docs, and time answering questions here - for free.  I'm here to answer specific "how to" questions.  If you have an issue that IS NOT from the samples, you need to follow the issue template here: https://github.com/shinyorg/shiny/issues/new/choose and provide a reproducible sample. 

I looked at the GPS sample this morning, it updated from the preview 2.3 nugets it was using to 2.4.1 without issue.  You need to do a little digging on your environment to figure things out.  Best of luck.
